crates/router/src/core/payments/routing.rs

```diff
@@ -1395,3 +1396,140 @@ pub async fn perform_success_based_routing(
         Ok(routable_connectors)
     }
 }
+
+/// elimination dynamic routing
+#[cfg(all(feature = "v1", feature = "dynamic_routing"))]
+pub async fn perform_elimination_routing(
+    state: &SessionState,
+    routable_connectors: Vec<api_routing::RoutableConnectorChoice>,
+    business_profile: &domain::Profile,
+    elimination_routing_configs_params_interpolator: routing::helpers::DynamicRoutingConfigParamsInterpolator,
+) -> RoutingResult<Vec<api_routing::RoutableConnectorChoice>> {
+    let dynamic_routing_algo_ref: api_routing::DynamicRoutingAlgorithmRef = business_profile
+        .dynamic_routing_algorithm
+        .clone()
+        .map(|val| val.parse_value("DynamicRoutingAlgorithmRef"))
+        .transpose()
+        .change_context(errors::RoutingError::DeserializationError {
+            from: "JSON".to_string(),
+            to: "DynamicRoutingAlgorithmRef".to_string(),
+        })
+        .attach_printable("unable to deserialize DynamicRoutingAlgorithmRef from JSON")?
+        .unwrap_or_default();
+
+    let elimination_algo_ref = dynamic_routing_algo_ref
+        .elimination_routing_algorithm
+        .ok_or(errors::RoutingError::GenericNotFoundError { field: "elimintaion_algorithm".to_string() })
+        .attach_printable(
+            "elimintaion_algorithm not found in dynamic_routing_algorithm from business_profile table",
```

Comment:
Correct the spelling in all places
```suggestion
            "elimination_algorithm not found in dynamic_routing_algorithm from business_profile table",
```